r/tests/testthat/test-RecordBatch.R

```diff
@@ -415,14 +414,42 @@ test_that("record_batch() handles null type (ARROW-7064)", {
   expect_equivalent(batch$schema,  schema(a = int32(), n = null()))
 })
 
-test_that("record_batch() scalar recycling", {
-  skip("Not implemented (ARROW-11705)")
+test_that("record_batch() scalar recycling with vectors", {
   expect_data_frame(
     record_batch(a = 1:10, b = 5),
     tibble::tibble(a = 1:10, b = 5)
   )
 })
 
+test_that("record_batch() scalar recycling with Scalars, Arrays, and ChunkedArrays", {
+  
+  expect_data_frame(
+    record_batch(a = Array$create(1:10), b = Scalar$create(5)),
+    tibble::tibble(a = 1:10, b = 5)
+  )
+  
+  expect_data_frame(
+    record_batch(a = Array$create(1:10), b = Array$create(5)),
+    tibble::tibble(a = 1:10, b = 5)
+  )
+  
+  expect_data_frame(
+    record_batch(a = Array$create(1:10), b = ChunkedArray$create(5)),
+    tibble::tibble(a = 1:10, b = 5)
+  )
+  
+})
+
+test_that("record_batch() no recycling with tibbles", {
```

Comment:
Why does this error, and should it?

Reply:
It does it because when we call `length()` on a `tibble`, it returns the number of columns, so in the case in the examples, 2. 

I don't know if it should - should we support recycling for tibbles? I guess we could - I suppose it's consistent with the other behaviour implemented.  This is definitely a problem though as if one of my tibbles has 1 column then weird things happen.

Reply:
I had a think about this, and `tibble::tibble` does support tibble recycling, so if we're trying to allow similar behaviour, I think it makes sense to implement it here.

Reply:
@nealrichardson I've now updated this so that the code uses either `length` or `nrow` depending on whether the argument is something that inherits from a `data.frame` or not.  

I got stuck trying to recycle tibbles as I couldn't think how to do it in a reasonable way with the packages we import.

Do you think there is one?  If not, I'm tempted to say it's out of the scope of this ticket anyway.